### PR TITLE
For new g6 firmware - clear LSR cache on any sensor stop or start

### PR DESCRIPTION
**Pull request by efidoman**
For any new g6 firmware transmitter, always clear the LSR calibration cache completely for any sensor start or stop. This will require at least one calibration for any sensor start or restart before Logger will process any glucose values using LSR or Single Point. Otherwise, Logger will wait for official glucose records after the 2-hour warm-up period.

### CODE REVIEW

File: xdrip-get-entries.sh [hunk 1]
```diff
@@ -502,6 +501,11 @@ function check_sensor_stop()
           echo "stopJSON = $stopJSON"
           # below done so that next time the egrep returns positive for this specific message and the log reads right
           echo "Already Processed Sensor Stop Message from Nightscout at $createdAt" >> ${LDIR}/nightscout-treatments.log
+          # Always clear LSR cache for any g6 start / stop
+          if [ "$txType" == "g6" ]; then
```
Comment (tynbendad) [inline xdrip-get-entries.sh]: i'd prefer to be able to turn this off by a xdripjs.json variable, so older g6 tx's can keep their calibration.

**Review by tynbendad — CHANGES_REQUESTED** [review]
would prefer to be able to turn off via xdripjs.json variable, the clearing on start/stop, for older g6 tx's.

File: xdrip-get-entries.sh (hunk 1 above)
Comment (efidoman) [reply xdrip-get-entries.sh]: Good point. I was able to use the version firmware to only do these clears for a g6 with new firmware version.

**Review by efidoman — COMMENTED** [review]
I think this is ready to go now. Any final suggestions / comments?

**Comment by tynbendad** [comment]
the commit looks great except for the one line i just commented on, looks like you used resetJSON instead of versionJSON?  i've also got a funny feeling about there being only one known 'new' firmware version number currently (just hard to believe) and how we will find out about new versions...

**Comment by efidoman** [comment]
There are multiple new firmware versions. The ones I know about so far are the following:

1.6.5.27
2.x.x.x

So, the function is checking for either of those using a wildcard for the 2.* firmware g6's. Make sense?

I'll check and update the resetJSON typo. Thanks for pointing that out.

**Comment by efidoman** [comment]
I'm going to fix the resetJSON typo in a new PR - it will work as is, but fails if someone does both reset and version requests in one 5 minute interval.

**Comment by tynbendad** [comment]
> There are multiple new firmware versions. The ones I know about so far are the following:
> 
> 1.6.5.27
> 2.x.x.x
> 
> So, the function is checking for either of those using a wildcard for the 2.* firmware g6's. Make sense?
> 
> I'll check and update the resetJSON typo. Thanks for pointing that out.

ah, thanks - i completely missed the "| 2.*".  that could be made into a check for >=2, but probably not close to getting a v3.* firmware yet anyway.

**Pull request merged**